jax/_src/interpreters/pxla.py

```diff
@@ -2000,7 +2000,7 @@ def are_all_shardings_default_mem_kind(da_object, shardings):
       return False
   return True
 
-memory_kind_propagate_rule = {}  # type: ignore
+memory_kind_propagate_rule: dict[Any, Any] = {}
```

Comment:
Why not type: ignore?

Reply:
`type: ignore` tells the type checker to treat this as `Any`, but we know it's at least a `dict`.

Reply:
But why did that fail on this PR where you are just bumping the version?

Reply:
Newer mypy emits an error when `memory_kind_propagate_rule` is modified.

Reply:
Hmm.. interesting. OOC, what's the error?